src/util-radix-tree.c

```diff
@@ -1009,6 +1009,12 @@ SCRadixNode *SCRadixAddKeyIPV4String(const char *str, SCRadixTree *tree, void *u
         SCRadixValidateIPv4Key((uint8_t *)&ip, netmask);
 #endif
     }
+
+    if (SCRadixFindKeyIPV4Netblock((uint8_t *)&ip, tree, netmask, NULL) != NULL) {
```

Comment:
I don't understand why this is done here. Adding duplicates is supported, right? It returns an existing node. In for example the IP only code we can have multiple sigs include the same ip/netblock.

Reply:
The idea here is to not add duplicates, that's why the check is done here.
But if we assume that having duplicate netblocks is valid, I'm afraid this approach is incorrect.

Reply:
> Adding duplicates is supported, right? It returns an existing node.

If the tree doesn't persist duplicates and returns an existing node, then the caller's `user` must still be handled properly.

If this is the case, then we must check if callers to `SCRadixAddKey*` can detect the case and free the `user` data, if appropriate.